scripts/signedbuildskadi.sh

```diff
@@ -1,274 +1,311 @@
-    printf '.'
-    sleep 5
-done
-echo "Setting the ElasticSearch default number of replicas to 0"
-
-curl -XPUT 'localhost:9200/_template/number_of_replicas' \
-    -d '{"template": "*","settings": {"number_of_replicas": 0}}' \
-    -H'Content-Type: application/json'
-
-echo "Waiting for Kibana service to respond to requests"
-until $(curl --output /dev/null --silent --head --fail http://localhost:5601); do
-    printf '.'
-    sleep 5
-done
-
-echo "Importing Saved Objects to Kibana and setting default index"
-curl -X POST "http://localhost:5601/api/saved_objects/_bulk_create" -H 'kbn-xsrf: true' -H 'Content-Type: application/json' --data-binary @/opt/Skadi/objects/kibana_6.x_cli_import.json
-curl -X POST "http://localhost:5601/api/kibana/settings/defaultIndex" -H "Content-Type: application/json" -H "kbn-xsrf: true" -d '{"value": "06876cd0-dfc5-11e8-bc06-31e345541948"}'
-
-# The TimeSketch container needs to be running before continuing and this
-# requires the other containers to be up and running too. This can take time
-# so this loop ensures all the parts are running and timesketch is responding
-# to web requets before continuing
-sudo docker restart timesketch
-echo "Waiting 10 seconds for TimeSketch to become available"
-sleep 10
-echo "Press CTRL-C at any time to stop installation"
-until $(curl --output /dev/null --silent --head --fail http://localhost/timesketch); do
-    echo "No response, restarting the TimeSketch container and waiting 10 seconds to try again"
-    sudo docker restart timesketch
-    sleep 10
-done
-echo "TimeSketch available. Continuing"
-
-# Install Grafana for Monitoring
-git clone https://github.com/orlikoski/skadi_dockprom.git
-cd skadi_dockprom
-
-# Write Grafana login creds to .env file
-echo ADMIN_USER=$GRAFANA_USER > ./.env
-echo ADMIN_PASSWORD=$GRAFANA_PASSWORD >> ./.env
-
-# This uses the docker-compose.yml found in the skadi_dockprom repo
-sudo docker-compose up -d
-
-# Installs and Configures CDQR and CyLR
-sudo -E bash /opt/Skadi/scripts/update.sh
-
-# Enable and Configure UFW Firewall
-echo "Enabling UFW firewall to only allow OpenSSH and Ngninx Full"
-sudo ufw allow 80
-sudo ufw allow 443
-sudo ufw allow 'OpenSSH'
-sudo ufw --force enable
-
-echo ""
-echo ""
-echo ""
-echo "Skadi Setup is Complete"
-echo ""
-echo "The Nginx reverse proxy setup and can be accessed at http://<IP Address> or http://localhost if installed locally:"
-if [ $default_skadi_passwords = "false" ]
-  then
-    echo "The following are the credentials needed to access this build and are stored in /opt/skadi_credentials if run-time generated credentials was chosen: "
-    echo ""
-    cat /opt/skadi_credentials
-    echo ""
-fi
-echo ""
-echo "The following files have credentials used in the build process stored in them:"
-echo "  - /opt/skadi_credentials (only if run-time generated credentials chosen)"
-echo "  - /opt/Skadi/Docker/.env"
-echo "  - /opt/Skadi/Docker/skadi_dockprom/.env"
+#!/bin/bash
+set -e
+

```

Comment:
All of the default values need to be at the top of the file or the functions can't use them

Reply:
Updated to be at top of file, will push a commit once all fixes are included